src/frequenz/sdk/data_ingestion/resampling/ring_buffer.py

```diff
@@ -0,0 +1,63 @@
+"""
+Ring buffer for storing ordered streamed component data for resampling purposes.
+
+Copyright
+Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+License
+MIT
+"""
+import logging
+from collections import deque
+from typing import Deque, Generic, List, TypeVar
+
+logger = logging.Logger(__name__)
+
+MAX_BUFFER_SIZE = 1000
+
+T = TypeVar("T")
+
+
+class RingBuffer(Generic[T]):
+    """Ring buffer for storing items."""
+
+    def __init__(
+        self,
+        size: int,
+    ) -> None:
+        """Initialize RingBuffer.
+
+        Args:
+            size: max number of items to be stored in the buffer
+
+        Raises:
+            ValueError: if provided `buffer_size` is smaller than 0 or greater than
+                `MAX_BUFFER_SIZE`
+        """
+        if size < 0 or size > MAX_BUFFER_SIZE:
+            raise ValueError(
+                f"Buffer size must be an integer in the (0, {MAX_BUFFER_SIZE}) range."
+            )
+
+        self._size = size
+        self._buffer: Deque[T] = deque()
+
+    @property
+    def items(self) -> List[T]:
+        """Get all the items in the ring buffer.
+
+        Returns:
+            All the items in the ring buffer.
+        """
+        return list(self._buffer)
```

Comment:
From the doc:
`deques support iteration, pickling, len(d), reversed(d), copy.copy(d), copy.deepcopy(d), `

So why you do this?

Reply:
I guess we can just remove it and call `list(ring_buffer)` everywhere instead.

Reply:
We will kill the separate class for the buffer so this can be marked as resolved.